### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by steveklabnik**
I want to start using scoped threads, but I also don't want to have to memorize the API :)

### CODE REVIEW

**Comment by aturon** [comment]
Thanks so much, @steveklabnik!

The only problem here is that the scoped spawning actually *does* still involve allocation under the hood: it dynamically tracks all of the cleanup it has to do when the scope is exited. But we can certainly make that more efficient than boxing.

**Comment by steveklabnik** [comment]
Ah ha! Okay, let me remove that line then.

To be clear, it has to allocate _something_, but it doesn't need to box up the things it's closing over? It's the cleanup stuff that gets allocated?

**Comment by aturon** [comment]
@steveklabnik That's the right idea, although the issue isn't so much the size as that the allocation can be amortized (e.g. the way we do with `Vec` where we alloc chunks in exponentially larger sizes as you run out of capacity).

Basically, the allocation is being managed by the library, which gives us leeway to be smarter about it.

**Comment by steveklabnik** [comment]
@aturon removed the allocation comment

File: src/scoped.rs [hunk 1]
```diff
@@ -78,6 +94,116 @@ impl<'a> Scope<'a> {
         });
     }
 
+    /// Create a scoped thread.
+    ///
+    /// `spawn` is similar to the [`spawn`][spawn] function in Rust's standard library. The
+    /// difference is that this thread is scoped, meaning that it's guaranteed to terminate
+    /// before the current stack frame goes away, allowing you to reference the parent stack frame
```
Comment (aturon) [inline src/scoped.rs]: Can you mention that this guarantee is ensured by having the parent thread join on the child thread before exiting the scope?

Comment (steveklabnik) [inline src/scoped.rs]: updated!

**Comment by aturon** [comment]
Merged, thanks again @steveklabnik!

**Comment by steveklabnik** [comment]
:confetti_ball:

**Pull request merged**